src/ipc/capnp/chain.capnp

```diff
@@ -0,0 +1,193 @@
+    checkChainLimits @25 (context :Proxy.Context, tx :Data) -> (result :Common.ResultVoid);
+    estimateSmartFee @26 (context :Proxy.Context, numBlocks :Int32, conservative :Bool, wantCalc :Bool) -> (calc :FeeCalculation, result :Data);
+    estimateMaxBlocks @27 (context :Proxy.Context) -> (result :UInt32);
+    mempoolMinFee @28 (context :Proxy.Context) -> (result :Data);
+    relayMinFee @29 (context :Proxy.Context) -> (result :Data);
+    relayIncrementalFee @30 (context :Proxy.Context) -> (result :Data);
+    relayDustFee @31 (context :Proxy.Context) -> (result :Data);
+    havePruned @32 (context :Proxy.Context) -> (result :Bool);
+    isReadyToBroadcast @33 (context :Proxy.Context) -> (result :Bool);
+    isInitialBlockDownload @34 (context :Proxy.Context) -> (result :Bool);
+    shutdownRequested @35 (context :Proxy.Context) -> (result :Bool);
+    initMessage @36 (context :Proxy.Context, message :Text) -> ();
+    initWarning @37 (context :Proxy.Context, message :Common.BilingualStr) -> ();
+    initError @38 (context :Proxy.Context, message :Common.BilingualStr) -> ();
+    showProgress @39 (context :Proxy.Context, title :Text, progress :Int32, resumePossible :Bool) -> ();
+    handleNotifications @40 (context :Proxy.Context, notifications :ChainNotifications) -> (result :Handler.Handler);
+    waitForNotificationsIfTipChanged @41 (context :Proxy.Context, oldTip :Data) -> ();
+    handleRpc @42 (context :Proxy.Context, command :RPCCommand) -> (result :Handler.Handler);
+    rpcEnableDeprecated @43 (context :Proxy.Context, method :Text) -> (result :Bool);
+    rpcRunLater @44 (context :Proxy.Context, name :Text, fn: RunLaterCallback, seconds: Int64) -> ();
+    getSetting @45 (context :Proxy.Context, name :Text) -> (result :Text);
+    getSettingsList @46 (context :Proxy.Context, name :Text) -> (result :List(Text));
+    getRwSetting @47 (context :Proxy.Context, name :Text) -> (result :Text);
+    updateRwSetting @48 (context :Proxy.Context, name :Text, update: SettingsUpdateCallback) -> (result :Bool);
+    overwriteRwSetting @49 (context :Proxy.Context, name :Text, value :Text, action :Int32) -> (result :Bool);
+    deleteRwSettings @50 (context :Proxy.Context, name :Text, action: Int32) -> (result :Bool);
+    requestMempoolTransactions @51 (context :Proxy.Context, notifications :ChainNotifications) -> ();
+    hasAssumedValidChain @52 (context :Proxy.Context) -> (result :Bool);
+}
+
+interface ChainNotifications $Proxy.wrap("interfaces::Chain::Notifications") {
+    destroy @0 (context :Proxy.Context) -> ();
+    transactionAddedToMempool @1 (context :Proxy.Context, tx :Data) -> ();
+    transactionRemovedFromMempool @2 (context :Proxy.Context, tx :Data, reason :Int32) -> ();
+    blockConnected @3 (context :Proxy.Context, role: UInt32, block :BlockInfo) -> ();
+    blockDisconnected @4 (context :Proxy.Context, block :BlockInfo) -> ();
+    updatedBlockTip @5 (context :Proxy.Context) -> ();
```

Comment:
Adding this here for context: I guess one the benefits of the `Notifications` and `NotificationsProxy` is avoiding raw pointer types in these interfaces. I went through the exercise of removing the pointer types from the validation interface before. Maybe this could eliminate the need for the logic in the `NotificationsProxy` in the first place. Not having to handle pointer types is also useful for the kernel API, where conveying the semantics of these pointers in the notifications to the using developer is tricky.